crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -327,13 +330,63 @@ async fn execute_ics20_transfer<S: StateWriteExt>(
     // if the asset is prefixed with `ibc`, the rest of the denomination string is the asset ID,
     // so we need to look up the full trace from storage.
     // see https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-001-coin-source-tracing.md#decision
-    if denom.prefix_is("ibc") {
+    if denom.prefix().starts_with("ibc") {
```

Comment:
This is upstream code, right? This is much nicer.

Reply:
no this is our code! but yeah this is better/ actually correct

Reply:
I see - yeah, that makes sense. Maybe we can remove the `prefix_is` method from core?